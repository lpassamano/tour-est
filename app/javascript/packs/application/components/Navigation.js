import React, { Component } from "react";
import { Link } from "@reach/router";

const Navigation = ({ currentStaffUser, onLogout }) => (
  <nav className="top-bar topbar-responsive">
    <div className="top-bar-title">
      <Link to="/admin" className="topbar-responsive-logo">
        <strong>tour-est</strong>
      </Link>
    </div>
    <div id="topbar-responsive" className="topbar-responsive-links">
      <div className="top-bar-right">
        {currentStaffUser ? (
          <ul className="menu simple vertical medium-horizontal">
            <li>
              <Link to="/admin">Home</Link>
            </li>
            <li>
              <Link to="/tours/new">Create Tour</Link>
            </li>
            <li>
              <a href="#" onClick={onLogout}>
                Log Out
              </a>
            </li>
          </ul>
        ) : (
          <ul className="menu simple vertical medium-horizontal">
            <li>
              <Link to="/login">Sign In</Link>
            </li>
            <li>
              <Link to="/sign-up">Create Account</Link>
            </li>
          </ul>
        )}
      </div>
    </div>
  </nav>
);

export default Navigation;
