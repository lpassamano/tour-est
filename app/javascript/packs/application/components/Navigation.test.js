import Navigation from "./Navigation";
import React from "react";
import { shallow } from "enzyme";

describe("<Navigation />", () => {
  test("render", () => {
    const onLogout = jest.fn();
    const currentStaffUser = { username: "leigh_p", cultural_center: "MoMA" };
    const component = shallow(
      <Navigation onLogout={onLogout} currentStaffUser={currentStaffUser} />
    );
    expect(component).toMatchSnapshot();
  });
});
