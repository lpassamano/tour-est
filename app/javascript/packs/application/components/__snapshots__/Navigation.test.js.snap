// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> render 1`] = `
<nav
  className="top-bar topbar-responsive"
>
  <div
    className="top-bar-title"
  >
    <[object Object]
      className="topbar-responsive-logo"
      id="tour-est"
      to="/"
    >
      <strong>
        tour-est
      </strong>
    </[object Object]>
  </div>
  <div
    className="topbar-responsive-links"
    id="topbar-responsive"
  >
    <div
      className="top-bar-right"
    >
      <ul
        className="menu simple vertical medium-horizontal"
      >
        <li>
          <a
            href="#"
            onClick={[MockFunction]}
          >
            Log Out
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
